scarb/src/compiler/compilers/test.rs

```diff
@@ -31,6 +34,23 @@ impl Compiler for TestCompiler {
                 && plugin.package.id.source_id == SourceId::for_std()
         });
 
+        let non_test_crate_ids = collect_non_test_crate_ids(&unit, db);
+
+        for crate_id in non_test_crate_ids {
+            if let Some(cfg) = db.crate_config(crate_id) {
+                db.set_crate_config(
+                    crate_id,
+                    Some(CrateConfiguration {
+                        root: cfg.root,
+                        settings: CrateSettings {
+                            cfg_set: Some(CfgSet::from_iter([Cfg::name("lib")])),
```

Comment:
Nit, while this works currently, it's not future-proof if we happen to want to add some extra `cfg` globally. Maybe it's worth to leave a comment about this problem here?

Reply:
Second this, would it be possible to iterate over the config and filter test cfg out?